### DISCUSSION TITLE
Row Level Security "Contains" Filter Clarification

### DISCUSSION THREAD

**Discussion by muellercornelius** [pocketbase/pocketbase#567] [Q&A] [answered]
Hi Guys ;),

I stumbled upon a problem with my API Security Rules Setup. 
I added to my users profile table a multiselect with four different roles. All of them can be active at once or only some.... 
Now I am wondering how the internal represantation of this "roles" field is and how it is converted when applying the filter.

My obvious idea was this:
`@request.user.profile.roles ~ 'user'`

or  another example: 
`@request.user.profile.roles ~ 'admin'`

This doesnt work sadly... Is my only option creating a second table for roles and relate the user with it?

Thank you for your help ;)

**Comment by ganigeorgiev** [comment, marked as answer]
The contains operators `~` and `!~` operators are similar to the SQL `LIKE` and `NOT LIKE` operators.

Hm, the above should work, but I think I have a bug when constructing the query for non-relation multi-select profiles field. Currently the non-relational multi-select field value is treated as plain string and you end up with something like:

```sql
'%["admin", "user"]%' LIKE '%users%'
--- instead of:
profile.roles LIKE '%users%'
```

I'll try to find time later today and will fix it in v0.7.5.

_btw. Plans for a proper `IN`-like operator is also in the roadmap and will be added in the near future._

**Reply by ganigeorgiev** [reply]
@muellercornelius This should be fixed now in [v0.7.5 release](https://github.com/pocketbase/pocketbase/releases/tag/v0.7.5). I've updated the tests to verify it, but if you are still experiencing issues, please let me know and I'll investigate it further.

**Reply by muellercornelius** [reply]
Tried it out and it works absolutely perfectly. Thank you very much ;).

**Reply by FlixMa** [reply]
Regarding the contains operator when both operands are table columns (instead of one multi-column field or another field of the related record) have a look at [this comment](https://github.com/pocketbase/pocketbase/discussions/662#discussioncomment-3761009) in another discussion about performance.
